src/libraries/System.Threading.Tasks.Dataflow/src/Base/DataflowBlock.cs

```diff
@@ -1381,6 +1381,34 @@ private object DebuggerDisplayContent
             object IDebuggerDisplay.Content { get { return DebuggerDisplayContent; } }
         }
         #endregion
+
+        #region ReceiveAllAsync
+
+#if NETSTANDARD2_1
+        /// <summary>Creates an <see cref="IAsyncEnumerable{T}"/> that enables receiving all of the data from the source.</summary>
+        /// <typeparam name="TOutput">Specifies the type of data contained in the source.</typeparam>
+        /// <param name="source">The source from which to asynchronously receive.</param>
+        /// <param name="cancellationToken">The <see cref="System.Threading.CancellationToken"/> which may be used to cancel the receive operation.</param>
+        /// <returns>The created async enumerable.</returns>
+        /// <exception cref="System.ArgumentNullException">The <paramref name="source"/> is null (Nothing in Visual Basic).</exception>
+        public static IAsyncEnumerable<TOutput> ReceiveAllAsync<TOutput>(this ISourceBlock<TOutput> source, CancellationToken cancellationToken = default)
+        {
+            // Validate arguments
+            if (source == null) throw new ArgumentNullException(nameof(source));
+
+            return source.ReceiveAllAsyncCore(cancellationToken);
+        }
+
+        private static async IAsyncEnumerable<TOutput> ReceiveAllAsyncCore<TOutput>(this ISourceBlock<TOutput> source, [EnumeratorCancellation] CancellationToken cancellationToken)
```

Comment:
You can make this a local function in ReceiveAllAsync.

Reply:
Done.